### DISCUSSION TITLE
Repeat the topic - Wrong sensor detected on the i2C bus --> AHT20 reported as I2c VEML6070 UV sensor

### DISCUSSION THREAD

**Discussion by MrPedalkin** [arendst/Tasmota#19515] [Support request]
Hello.
I'm trying to connect the AHT20+BMP280 sensor to Tasmata 13.1.0 "Sensors". The "status 0" command does not show the number 43 in the list of available drivers. As a result, the BMP280 sensor works, but the AHT 20 does not work. When I experimented, I disabled all the drivers, but it didn't help. What driver should there be for device 43?

P.s. Subject https://github.com/arendst/tasmota/issues/18428 I've already watched it.
```
03:33:30.369 CMD: GPIO 255
03:33:30.380 MQT: stat/sonoff_th10/RESULT = {"GPIO0":{"32":"Button1"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"608":"I2C SCL"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"224":"Relay1"},"GPIO13":{"320":"Led_i1"},"GPIO14":{"640":"I2C SDA"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}
``` 
```
02:16:27.706 CMD: i2cscan
02:16:27.728 MQT: stat/sonoff_th10/RESULT = {"I2CScan":"Device(s) found at 0x38 0x77"}
``` 
```
02:17:31.104 CMD: status 0
02:17:31.112 MQT: stat/sonoff_th10/STATUS = {"Status":{"Module":4,"DeviceName":"Sonoff","FriendlyName":["Sonoff"],"Topic":"sonoff_th10","ButtonTopic":"0","Power":1,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
02:17:31.119 MQT: stat/sonoff_th10/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"sonoffs","OtaUrl":"https://ota.tasmota.com/tasmota/release/tasmota-sensors.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:31:46","StartupUTC":"2023-09-14T00:45:45","Sleep":50,"CfgHolder":4617,"BootCount":4190,"BCResetTime":"2023-08-09T05:51:37","SaveCount":4369,"SaveAddress":"FB000"}}
02:17:31.123 MQT: stat/sonoff_th10/STATUS2 = {"StatusFWR":{"Version":"13.1.0(sensors)","BuildDateTime":"2023-08-15T07:59:54","Boot":7,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"355/699"}}
02:17:31.127 MQT: stat/sonoff_th10/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["New",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A64002800000000","00000080","00006000","00004000","00000000"]}}
02:17:31.145 MQT: stat/sonoff_th10/STATUS4 = {"StatusMEM":{"ProgramSize":700,"Free":300,"Heap":16,"ProgramFlashSize":1024,"FlashSize":1024,"FlashChipId":"1420C2","FlashFrequency":40,"FlashMode":"DOUT","Features":["00000809","8F9A8787","0415A005","B7F7BFCF","05DA9BC4","64367CC7","00084052","20000000","54000020","0008C081"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,14,16,17,20,21,24,29,34,62,65,66,68","Sensors":"1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,17,18,19,20,21,22,26,31,34,37,39,40,42,43,45,51,52,55,56,58,59,62,64,66,67,74,98,103,105","I2CDriver":"!7,!8,!9,!10,!11,!12,!13,!14,!15,!17,!18,!20,!24,!29,!31,!36,!41,!42,!44,!46,!48,!69,!76"}}
02:17:31.153 MQT: stat/sonoff_th10/STATUS5 = {"StatusNET":{"Hostname":"sonoff-th10-2216","IPAddress":"192.168.254.251","Gateway":"192.168.254.254","Subnetmask":"255.255.255.0","DNSServer1":"172.16.1.1","DNSServer2":"0.0.0.0","Mac":"CC:50:E3:52:E8:A8","Webserver":2,"HTTP_API":1,"WifiConfig":2,"WifiPower":17.0}}
02:17:31.159 MQT: stat/sonoff_th10/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.19.2","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_52E8A8","MqttUser":"AndyMQTT","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
02:17:31.165 MQT: stat/sonoff_th10/STATUS7 = {"StatusTIM":{"UTC":"2023-09-14T01:17:31","Local":"2023-09-14T02:17:31","StartDST":"2023-03-26T02:00:00","EndDST":"2023-10-29T03:00:00","Timezone":"+01:00","Sunrise":"06:25","Sunset":"19:06"}}
02:17:31.172 MQT: stat/sonoff_th10/STATUS10 = {"StatusSNS":{"Time":"2023-09-14T02:17:31"}}
02:17:31.178 MQT: stat/sonoff_th10/STATUS11 = {"StatusSTS":{"Time":"2023-09-14T02:17:31","Uptime":"0T00:31:46","UptimeSec":1906,"Heap":16,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"ON","Wifi":{"AP":1,"SSId":"New","BSSId":"4E:5E:0C:52:E7:D7","Channel":3,"Mode":"11n","RSSI":100,"Signal":-36,"LinkCount":1,"Downtime":"0T00:00:03"}}}
```

**Comment by HofeBY** [comment]
"i2cdriver" don´t show driver 43, so you have to compile your own version of "tasmota.bin" with included driver "AHT20"

**Reply by MrPedalkin** [reply]
Thanks for the reply. Is there a link to a discussion on how to do this? I'm studying the topic https://tasmota.github.io/docs/PlatformIO/#install-platformio and I'm trying to do as written, but I can't figure out where to disable unused drivers to save flash memory.

**Reply by sfromis** [reply]
https://tasmota.github.io/docs/Compile-your-build/#enabling-a-feature-in-tasmota

In the real world, it is a waste of time to cut down on flash memory usage, unless you have other things to add instead, and the total size getting "too large".

**Reply by HofeBY** [reply]
@MrPedalkin
For Windows Systems:
Download&install https://code.visualstudio.com/
In VSC add extension SW "PlatformIO IDE"
download&uncompress Binary from https://github.com/arendst/Tasmota
first compile tasmota(ESP8285/ESP8266) and tasmota32(ESP32) will download additional Software & build environment
.
study ...\tasmota\my_user_config.h for possible Drivers & also unwated Drivers !
.
add wished Driver support in  ...\tasmota\user_config_override.h
.
remove unwished Drivers in ...\tasmota\include\tasmota_configurations.h (ESP8285/ESP8266) or tasmota_configurations_ESP32.h (ESP32) mybe Driver by uncomment "#define USE_VEML6070" to "//#define USE_VEML6070"
or disable driver 12 later by command "I2CDriver12 0" 
.
Now build "tasmota" or "tasmota32" depending to your CPU
Install ...\build_output\firmware\tasmota.bin or tasmota32.bin to your device
.
done

**Reply by ccc-dimla** [reply]
Hello, I wanted to get an AHT20 + BMP280 to work with a d1 Mini esp8266 and tasmota. Temperature and air pressure are displayed correctly, but the humidity is wrong. Here it says something about veml6070 UV index. I came across a similar problem, here in the my_user_config.h #define USE_BMP and #define USE_ATH2x were commented out. Unfortunately without success, the value is still displayed incorrectly. What am I doing wrong?

**Reply by sfromis** [reply]
As indicated on https://tasmota.github.io/docs/I2CDEVICES/ it is recommended to disable other drivers potentially using clashing I2C addresses. That is relevant to driver index 12 for VEML6070, as I assume that you do not have this sensor.

**Comment by mherweg** [comment]
this helped in my case:
enter in tasmota console:
`
I2cDriver12 0    
`
  (enter)

(Tasmota14.5.0.2 (1b51aef-tasmota32_2M) on ESP32C3 )
